tests/test_mc_comm.py

```diff
@@ -90,6 +159,79 @@ def test_McCommunicationProcess_set_board_connection__sets_connect_to_mc_simulat
     assert actual[3] is None
 
 
+@freeze_time("2021-03-15 13:05:10.121212")
+def test_McCommunicationProcess_create_connections_to_all_available_boards__populates_connections_list_with_a_serial_object_when_com_port_is_available__and_sends_correct_message_to_main(
+    four_board_mc_comm_process, mocker, patch_comports, patch_serial_connection
+):
+    comport, mocked_comports = patch_comports
+    dummy_serial_obj, mocked_serial = patch_serial_connection
+    mc_process = four_board_mc_comm_process["mc_process"]
+    board_queues = four_board_mc_comm_process["board_queues"]
+    mocker.patch.object(
+        mc_process,
+        "determine_how_many_boards_are_connected",
+        autospec=True,
+        return_value=1,
+    )
+    board_idx = 0
+
+    mc_process.create_connections_to_all_available_boards()
+    confirm_queue_is_eventually_of_size(board_queues[0][1], 1)
+    assert mocked_comports.call_count == 1
+    assert mocked_serial.call_count == 1
+    actual_connections = mc_process.get_board_connections_list()
+    assert actual_connections[1:] == [None, None, None]
+    actual_serial_obj = actual_connections[board_idx]
+    assert isinstance(actual_serial_obj, Serial)
+    assert mocked_serial.call_args_list[0][1] == {
+        "port": comport,
+        "baudrate": SERIAL_COMM_BAUD_RATE,
+        "bytesize": 8,
+        "timeout": 0,
+        "stopbits": serial.STOPBITS_ONE,
+    }
+
+    actual_message = board_queues[0][1].get(timeout=QUEUE_CHECK_TIMEOUT_SECONDS)
+    assert actual_message["communication_type"] == "board_connection_status_change"
+    assert actual_message["board_index"] == board_idx
+    assert actual_message["is_connected"] is True
+    assert actual_message["timestamp"] == "2021-03-15 13:05:10.121212"
+
+
+@freeze_time("2021-03-15 13:05:10.121212")
```

Comment:
note for future: typically, we like to freeze the time at slightly different values in different tests, just to inject some variety and ensure that somehow a time value wasn't hardcoded in the main code